lib/src/lifecycle_module.dart

```diff
@@ -52,12 +54,21 @@ enum LifecycleState {
 /// Intended to be extended by most base module classes in order to provide a
 /// unified lifecycle API.
 abstract class LifecycleModule extends SimpleModule with Disposable {
+  static int _nextId = 0;
+  // Used by tracing to tell apart multiple instances of the same module
```

Comment:
Looks like this and other comments need to make their way into the lifecycle tracing PR

Reply:
Is it alright that this PR is already merging them?

Reply:
It would be nice if they were in the other PR so that work could be reviewed atomically, but it's fine it it's too much effort to make that happen.